catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java

```diff
@@ -0,0 +1,378 @@
+        for (SaxEventHandler transformer : eventHandlers) {
+            transformer.characters(ch, start, length);
+        }
+    }
+
+    /**
+     * Takes in a sax event from {@link SaxEventHandlerDelegate#parser}
+     * and passes it to the {@link SaxEventHandlerDelegate#eventHandlers}
+     *
+     * @param namespaceURI the namespaceURI that is passed in by {@link SaxEventHandlerDelegate}
+     * @param localName    the localName that is passed in by {@link SaxEventHandlerDelegate}
+     * @param qName        the qName that is passed in by {@link SaxEventHandlerDelegate}
+     * @throws SAXException
+     */
+    @Override
+    public void endElement(String namespaceURI, String localName, String qName)
+            throws SAXException {
+
+        for (SaxEventHandler transformer : eventHandlers) {
+            transformer.endElement(namespaceURI, localName, qName);
+        }
+
+    }
+
+    /**
+     * Takes in a sax event from {@link SaxEventHandlerDelegate#parser}
+     * and passes it to the {@link SaxEventHandlerDelegate#eventHandlers}
+     *
+     * @param prefix the prefix that is passed in by {@link SaxEventHandlerDelegate}
+     * @param uri    the uri that is passed in by {@link SaxEventHandlerDelegate}
+     * @throws SAXException
+     */
+    @Override
+    public void startPrefixMapping(String prefix, String uri) throws SAXException {
+        for (SaxEventHandler transformer : eventHandlers) {
+            transformer.startPrefixMapping(prefix, uri);
+        }
+    }
+
+    private void configureEventHandlerLookup() {
+
+        for (SaxEventHandler eventHandler : eventHandlers) {
+            if (eventHandler.getWatchedElements() != null) {
+                for (String element : eventHandler.getWatchedElements()) {
+                    List<SaxEventHandler> tmpList = eventHandlerLookup.get(element) != null ?
+                            eventHandlerLookup.get(element) :
+                            new ArrayList<>();
+                    tmpList.add(eventHandler);
+                    eventHandlerLookup.put(element, tmpList);
+                }
+            }
+        }
+
+    }
+
+    public SaxEventHandlerDelegate setMetacardType(MetacardType metacardType) {
+        this.metacardType = metacardType;
+        return this;
+    }
+
+    public TeeInputStream getMetadataStream(InputStream inputStream, OutputStream outputStream) {
+        return new TeeInputStream(inputStream, outputStream);
+    }
+
+}
+
+/**
+ * A private class used to handle errors that occur during SAX Parsing. It allows all the errors
+ * that occur during the parsing of a single document to be easily and succinctly logged at the end
+ * of parsing.
+ */
+class InputTransformerErrorHandler implements ErrorHandler {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(InputTransformerErrorHandler.class);
+
+    private ByteArrayOutputStream out;
```

Comment:
It's not a big deal that these streams are not closed. I expect that Java will close them automatically when the object is garbage collection. If the stream were open on files, then explicitly closing them might be necessary to prevent a resource leak.